dropwizard-jersey/src/main/java/io/dropwizard/jersey/guava/OptionalParamConverterProvider.java

```diff
@@ -0,0 +1,72 @@
+package io.dropwizard.jersey.guava;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.jersey.internal.inject.Providers;
+import org.glassfish.jersey.internal.util.ReflectionHelper;
+import org.glassfish.jersey.internal.util.collection.ClassTypePair;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Set;
+
+@Singleton
+public class OptionalParamConverterProvider implements ParamConverterProvider {
+
+    private final ServiceLocator locator;
+
+    @Inject
+    public OptionalParamConverterProvider(final ServiceLocator locator) {
+        this.locator = locator;
+    }
+
+    @Override
+    public <T> ParamConverter<T> getConverter(final Class<T> rawType, final Type genericType, final Annotation[] annotations) {
```

Comment:
I think there should be a check whether it's really an `Optional` (e.g. `if (!Optional.class.equals(rawType)) { return null; }`). Right now, a converter is returned for any generic that has a type parameter for which some converter exists, resulting in a `ClassCastException` when being used (lines 37 and 54).